extensions-core/kafka-indexing-service/src/test/java/org/apache/druid/indexing/kafka/KafkaIndexTaskClientTest.java

```diff
@@ -1054,6 +1054,36 @@ public void testSetEndOffsetsAsyncWithResume() throws Exception
     }
   }
 
+  @Test
+  public void testCancelTaskPauseRequests() throws Exception
+  {
+    final int numRequests = TEST_IDS.size();
+    Capture<Request> captured = Capture.newInstance(CaptureType.ALL);
+    EasyMock.expect(responseHolder.getStatus()).andReturn(HttpResponseStatus.OK).anyTimes();
+    EasyMock.expect(responseHolder.getContent()).andReturn("{\"0\":\"1\"}").anyTimes();
+    EasyMock.expect(httpClient.go(
+            EasyMock.capture(captured),
+            EasyMock.anyObject(ObjectOrErrorResponseHandler.class),
+            EasyMock.eq(TEST_HTTP_TIMEOUT)
+    )).andReturn(
+            okResponseHolder()
+    ).times(numRequests);
+    replayAll();
+
+    List<ListenableFuture<Map<Integer, Long>>> futures = new ArrayList<>();
+    for (String testId : TEST_IDS) {
+      futures.add(client.pauseAsync(testId));
+    }
+
+    Assert.assertEquals(client.getPauseFutureSize(), TEST_IDS.size());
+
+    Futures.allAsList(futures).get();
+    verifyAll();
+
+    client.cancelTaskPauseRequests();
+    Assert.assertEquals(client.getPauseFutureSize(), 0);
```

Comment:
This assertion can lead to flaky tests as it is possible that at this point some of the pauses would have already succeeded (or failed) thus changing the value returned by `getPauseFuturesSize`.

Added some suggestions in `SeekableStreamTaskClient` to rewrite these tests.

Reply:
removed